specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,422 @@
+  /// Save the page as mhtml
+  COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_SINGLE_FILE,
+  /// Save the page as html, plus, download the page related source files 
+  /// (for example CSS, JavaScript, images, and so on) in a directory with 
+  /// the same filename prefix.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_COMPLETE,
+} COREWEBVIEW2_SAVE_AS_REQUESTED_KIND;
+
+/// Status of a programmatic save as call, indicates the result
+/// for method `RequestSaveAs`
+[v1_enum] typedef enum COREWEBVIEW2_SAVE_AS_REQUESTED_RESULTS {
+  /// Could not perform Save As because the destination file path is an invalid path.
+  ///
+  /// It is considered as invalid when:
+  /// the path is empty or a relativate path, the parent directory doesn't
+  /// exist, or the path is a directory.
+  ///
+  /// Parent directory can be itself, if the path is root directory, or
+  /// root disk. When the root doesn't exist, the path is invalid.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_INVALID_PATH,
+  /// Could not perform Save As because the destination file path already exists and 
+  /// replacing files was not allowed by the `AllowReplace` property.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_FILE_ALREADY_EXISTS,
+  /// Could not perform Save As when the `Kind` property selection not
+  /// supported because of the content MIME type or system limits
+  ///
+  /// MIME type limits please see the emun `COREWEBVIEW2_SAVE_AS_REQUESTED_KIND`
+  ///
+  /// System limits might happen when select `HTML_ONLY` for an error page,
+  /// select `COMPLETE` and WebView running in an App Container, etc.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_NOT_SUPPORTED,
+  /// Did not perform Save As because the end user cancelled or the 
+  /// CoreWebView2SaveAsRequestedEventArgs.Cancel property was set to TRUE.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_CANCELLED,
+  /// The RequestSaveAs method call completed successfully and the download has started.
+  COREWEBVIEW2_SAVE_AS_REQUESTED_COMPLETED,
+} COREWEBVIEW2_SAVE_AS_REQUESTED_RESULTS;
+
+[uuid(15e1c6a3-c72a-4df3-91d7-d097fbec3bfd), object, pointer_default(unique)]
+interface ICoreWebView2_20 : IUnknown {
+  /// Programmatically trigger a save as action for the current top-level document. 
+  /// The `SaveAsRequested` event will be raised.
+  ///
+  /// Opens a system modal dialog by default. If it was already opened, this method 
+  /// would not open another one. If the `SuppressDefaultDialog` is TRUE, won't open 
+  /// the system dialog.
+  ///
+  /// The method can return a detailed info to indicate the call's result. 
+  /// Please see COREWEBVIEW2_SAVE_AS_REQUESTED_RESULTS
+  ///
+  /// \snippet ScenarioSaveAs.cpp ProgrammaticSaveAs
+  HRESULT RequestSaveAs([in] ICoreWebView2RequestSaveAsCompletedHandler* handler);
+
+  /// Add an event handler for the `SaveAsRequested` event. This event is raised
+  /// when save as is triggered, programmatically or manually.
+  ///
+  /// \snippet ScenarioSaveAs.cpp AddEventHandler
+  HRESULT add_SaveAsRequested(
```

Comment:
To go with renaming the method to ShowSaveAsUI, we rename this event as well: 

SaveAsUIShowing